draft-ietf-gnap-core-protocol.md

```diff
@@ -2368,108 +2370,199 @@ unknown_request
-          requested resources and direct information
-
-- allow the RO to modify the parameters of the request (such as
-          disallowing some requested resources or specifying an account or
-          record)
-
-- provide warnings to the RO about potential attacks or negative
-    effects of the requested information
-
-## Interaction at a Redirected URI {#interaction-redirect}
+# Determining Authorization and Consent {#authorization}
+
+When the client instance makes its [request](#request) to the AS for delegated access, it 
+asks for two kinds of information in response:
+
+- the access being requested in the `access_token` request parameter
+- the subject information being requested in the `subject` request parameter
+- any additional requested information defined by extensions of this protocol
+
+The AS determines what authorizations and consents are required to fulfill this requested delegation. The details of how the
+AS makes this determination are out of scope for this document. However, there are several common patterns
+defined and supported by GNAP for fulfilling these requirements, including information send by the client instance, information
+gathered through the interaction process, and information supplied by external parties. An individual AS
+can define its own policies and processes for deciding when and how to gather the necessary authorizations
+and consent.
+
+The client instance can supply information indicating that the client instance has the
+right to be granted the requested delegation immediately, including:
+
+- the identity of the client instance, known from the presented keys or associated identifiers
+- the identity of the end user presented in the `user` request parameter
+- any additional information presented by the client instance in the request, including any extensions
+
+The AS will verify this presented information in the context of the client instance's request and
+can only trust the information as much as it trusts the presentation and context of the information.
+If the AS determines that the information presented in the initial request is sufficient for granting the requested
+access, the AS MAY return the positive results immediately in its [response](#response) with
+access tokens and subject information.
+
+If the AS determines that additional runtime authorization is required, the AS can either deny the request
+outright or use a number of means at its disposal to gather that authorization from the appropriate ROs, 
+including for example:
+
+- starting interaction with the end user through the client instance
+- challenging the client instance through a response mechanism
+- request that the client instance present specific additional information
+- contacting a RO through an out-of-band mechanism
+- contacting an auxiliary software process through an out-of-band mechanism
+- any additional mechanisms defined by extensions to this protocol
```

Comment:
I get the idea of that last bullet point, the problem however is that it's not entirely clear what is core and what is an extension, even if the bullet items just above